pennylane/wires.py

```diff
@@ -354,6 +359,12 @@ def all_wires(list_of_wires):
 
             combined.extend(wire for wire in wires.labels if wire not in combined)
 
+        if sort:
+            if all([isinstance(w, int) for w in combined]):
+                combined = sorted(combined)
+            else:
+                combined = sorted(combined, key=str)
```

Comment:
Yes pretty much it. The order like you said doesn't really matter as long as it's unique. The special int treatment here is more because of what I thought a user's natural expectation for things to behave is, especially in the context of converting OpenFermion's ints to our ints. Debugging this behaviour is not the most trivial thing if the user expect things to be in order.